### PR TITLE
fix(test): update library causing flaky test behavior

### LINKED ISSUE #1123: Flaky test fix hangs test suite
## Describe the bug
Flaky test `com.alipay.sofa.common.boot.logging.test.DefaultLevelTest#testDefaultLevel` reported by the UIUC tool nondex. The particular testcase crashes due to multiple instances of the testcase fork trying to write to the same stream (as far as I understand):

Command: `mvn edu.illinois:nondex-maven-plugin:2.1.1:nondex -pl sofa-boot-project/sofa-boot-starters/log-sofa-boot-starter -DnondexRuns=10 -Dtest=com.alipay.sofa.common.boot.logging.test.DefaultLevelTest#testDefaultLevel`

Output:
```shell
Corrupted STDOUT by directly writing to native stream in forked JVM 1. Stream 'Error occurred during initialization of VM'.
java.lang.IllegalArgumentException: Stream stdin corrupted. Expected comma after third character in command 'Error occurred during initialization of VM'.
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient$OperationalData.<init>(ForkClient.java:507)
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient.processLine(ForkClient.java:210)
	at org.apache.maven.plugin.surefire.booterclient.output.ForkClient.consumeLine(ForkClient.java:177)
	at org.apache.maven.plugin.surefire.booterclient.output.ThreadedStreamConsumer$Pumper.run(ThreadedStreamConsumer.java:88)
``` 
This is a known issue with the surefire plugin, mentioned in this [StackOverflow](https://stackoverflow.com/questions/55272870/surefire-maven-plugin-corrupted-stdout-by-directly-writing-to-native-stream-in) thread. 

As per the thread, I updated the plugin version to `3.0.0-M5`, and all existing testcases pass for `mvn test`, but the original command for nondex hangs on the second test fork, with only the first one returning an output. Please advise.

## To Reproduce
1. Update version for the `maven.surefire.plugin` to '3.0.0-M5' from '2.22.2' in both occurences
2. Add the `<forkNode implementation="org.apache.maven.plugin.surefire.extensions.SurefireForkNodeFactory"/>` line at line 436 in file `.../sofa-boot/sofa-boot-project/sofa-boot-parent/pom.xml`
3. Clean and re-build project
4. Run nondex command
5. See error

## Expected behavior
Testcase should always pass, and not exhibit flaky behavior. Concurrent testcases should not hang on a shared resource.

## Screenshots
N/A

## Environment:
- Project Version [e.g. SOFA-RPC 1.0.0] : 3.17.0
- Language Version (e.g. Go 1.13): OpenJDK 11
- OS version (e.g. `uname -a`):  Linux Desktop-XXX 5.15.90.1-microsoft-standard-WSL2 SMP Fri Jan 27 02:56:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
- IDE version: VSCode-latest

### PR DESCRIPTION
**Pull request by msalman-abid**
Bumped version for the Maven surefire plugin, where concurrent JVM runs for testcases in JDK 11 were causing race conditions (and therefore flaky test failures).

Flaky test evidence using the NonDex tool: https://github.com/msalman-abid/sofa-boot/actions/runs/4457234377 

Closes #1123

### CODE REVIEW

**Comment by msalman-abid** [comment]
@HzjNeverStop unable to tag you as reviewer, please check if this looks ok

File: sofa-boot-project/sofa-boot-parent/pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
         <maven.source.plugin>3.0.0</maven.source.plugin>
         <maven.javadoc.plugin>2.9.1</maven.javadoc.plugin>
         <jacoco.maven.plugin>0.8.7</jacoco.maven.plugin>
-        <maven.surefire.plugin>2.22.2</maven.surefire.plugin>
+        <maven.surefire.plugin>3.0.0-M6</maven.surefire.plugin>
```
Comment (HzjNeverStop) [inline sofa-boot-project/sofa-boot-parent/pom.xml:30]: Could we use 3.0.0 release version instead?

Comment (msalman-abid) [reply sofa-boot-project/sofa-boot-parent/pom.xml]: Addressed. The release version was failing install attempts locally, but it seems to be sorted now.

**Review by HzjNeverStop — APPROVED** [review]
LGTM

**Pull request merged**